tools/build/src/devfile/meta-yaml-to-devfile-yaml.ts

```diff
@@ -131,13 +131,14 @@ export class MetaYamlToDevfileYaml {
           devfileEndpoint.attributes['type'] = 'main';
         }
 
-        // if it's secured, remove secure option for now but add extra s on the protocol
+        // if it's secured, remove 'secure' option for now but change protocol to secure (https or wss)
         if (devfileEndpoint.attributes && devfileEndpoint.attributes.secure === true) {
           devfileEndpoint.secure = false;
```

Comment:
I don't understand why we change **secure** to false when it was set to true.
I think we shouldn't change the value.

Reply:
@l0rd do you happen to remember why it was done that way? I recall that `secure` attribute might be in general not supported by DWO

Reply:
No, I suspect that `secured` is currently ignored (it was used by che-server when it was creating routes) and to have a secured endpoint (that require authentication) you need to set the `supportURLRewrite` to true.

Reply:
The Devfile API docs give

> Describes whether the endpoint should be secured and protected by some authentication process. This requires a protocol of https or wss.

but I think (haven't verified) that this field is either ignored in Che or is interpreted as "use https instead of http" (which isn't really appropriate, API-wise). I think the "most correct" solution would be something like

* If `secured` and `supportURLRewrite`, use gateway URL
* If either is false, use plain route
* If `secured` but not `supportURLRewrite`, fail workspace with error as we can't put auth on a plain route

Reply:
@l0rd @amisevsk I would probably leave it as-is for now (just fixing `httpss / wsss`cases), and I think we need a separate issue for supporting `secure` attribute properly  and not do any weird patching on plugin-registry end. Please, approve if this makes sense